packages/typespec-vscode/src/typespec-log-output-channel.ts

```diff
@@ -0,0 +1,108 @@
+import vscode, { LogOutputChannel } from "vscode";
+
+const TRACE_PREFIX = /^\[Trace.*?\] /iu;
+const DEBUG_PREFIX = /^\[Debug.*?\] /iu;
+const INFO_PREFIX = /^\[Info.*?\] /iu;
+const WARN_PREFIX = /^\[Warn.*?\] /iu;
+const ERROR_PREFIX = /^\[Error.*?\] /iu;
+
+export class TypeSpecLogOutputChannel implements LogOutputChannel {
```

Comment:
I see the discussion on creating this but what is the spec for this, how do we know it will keep working

Reply:
I think the LogOutputChannel is interface exposed by vscode to end user which would be backward compatible. As for the keywords "Info", "Error", "Warn"..., I didn't find any official doc to mention them, but I think it should be fine to use them here because
1. These keywords have been there for a very long time (~7 years) and all the trace/log messages are customer facing, which means it's very likely that many customers have depended on these keywords to do some parsing, so I dont think vscode would change them without a very good reason, which may cause many potential breaks to their customers. And I really can't think about a 'good reason' to change there keywords...
2. If vscode want to do some change here to make RemoteConsole and LogOutputChannel work together, it's likely they will change to call the outputchannel.info/debug/error/... directly instead of append/appendline with refined string, so the current implementation should still work.
3. In worst case, if vscode does make a break change in these keywords for some reason (really can't think one). The impact to us is still very limited because the default config for trace and log is 'off' and 'info', and even when user turn on more traces and logs, they can still see the log messages, just the loglevel will be marked as invalid. (Also we should be able to make a fix to it very quickly). 
So i think it's fine for us to take this workaround here. thanks.